### PR TITLE
sig-cluster-lifecycle: move logical-cluster to a subproject repo

### PR DESCRIPTION
**Pull request by neolit123**
As the [logical-cluster repository README](https://github.com/kubernetes-sigs/logical-cluster) indicates, this project was primary created for serving in CAPI usage. Later it can become used by other subprojects like kOps, or cloud providers.

However, having it as a subproject creates extra work for us that is not necessary - e.g. subproject yearly reports. After discussion in the SIG meeting of 31.10.2023, we have agreed to move it from a subproject to a "associated repository" of the CAPI subproject.

A similar case for this can be seen under kubeadm - e.g. see k/system-validators.

**Which issue(s) this PR fixes**:
NONE

### CODE REVIEW

**Comment by neolit123** [comment]
/cc @vincepri @fabriziopandini @justinsb 
PTAL for lgtm. needs approval form k/c maintainers.

/kind cleanup

**Comment by neolit123** [comment]
/hold

> https://github.com/kubernetes/community/blob/master/OWNERS#L13C1-L13C27

> Once this PR has been reviewed and has the lgtm label, please assign [pacoxu](https://github.com/pacoxu) for approval.

/assign @pacoxu 
seems like you can help with approval as steering member :)

**Review by pacoxu — APPROVED** [review]
/lgtm

**Comment by neolit123** [comment]
> Hold can be cancelled now?

yep.

/hold cancel

**Pull request merged**
